src/olympia/lib/akismet/models.py

```
@@ -0,0 +1,111 @@
+    reported = models.BooleanField(default=False)
+
+    # Fks to the supported types of content we support submitting
+    # Just Rating at first
+    rating_instance = models.ForeignKey(
+        'ratings.Rating', related_name='+', null=True,
+        on_delete=models.SET_NULL)
+
+    class Meta:
+        db_table = 'akismet_reports'
+
+    def _get_data(self):
+        data = {
+            'blog': settings.SITE_URL,
+            'user_ip': self.user_ip,
+            'user_agent': self.user_agent,
+            'referrer': self.referrer,
+            'permalink': self.content_link,
+            'comment_type': self.comment_type,
+            'comment_author': self.user_name,
+            'comment_author_email': self.user_email,
+            # 'comment_author_url': '',
+            'comment_content': self.comment,
+            'comment_date_gmt': self.comment_modified,
+            'comment_post_modified_gmt': self.content_modified,
+
+            # 'blog_lang' : get_language(),
+            'blog_charset': 'utf-8',
+            # 'user_role': 'administrator' if user.is_staff else ''
+            'is_test': not settings.AKISMET_REAL_SUBMIT,
+        }
+        return {key: value for key, value in data.items() if value is not None}
+
+    def comment_check(self):
+        url = settings.AKISMET_API_URL.format(
+            api_key=settings.AKISMET_API_KEY, action='comment-check')
+        response = requests.post(
+            url, data=self._get_data(), headers=self.HEADERS,
+            timeout=settings.AKISMET_API_TIMEOUT)
+        try:
+            outcome = response.json()
+            discard = response.headers.get('X-akismet-pro-tip') == 'discard'
+            if outcome is True or outcome is False:
+                self.update(result=(
+                    self.HAM if not outcome else
+                    self.DEFINITE_SPAM if discard else self.MAYBE_SPAM))
+                return self.result
+            # If outcome isn't True or False it's invalid.
```

That comment seems weirdly indented.

---

What are other possible outcomes? ~~Are they worth logging instead of silently ignoring?~~ (Just noted there's a logging statement just below)

---

It was indented in line with the except but I suppose it's more relevant just before the return

---

https://github.com/mozilla/addons-server/pull/9127/commits/cdadcdde76edd177d6480da60cf55544b9258eba